docs/source/configuration/authorization.mdx

```diff
@@ -639,12 +639,17 @@ authorization:
 
 This option will entirely reject queries that would have some parts filtered by authorization directives. The response will contain the list of paths that are affected.
 
-### log_errors
+### errors
 
 ```yaml title="router.yaml"
 authorization:
   preview_directives:
-    log_errors: true
+    errors:
+      log: true # default: true
+      response: "errors" # possible values: "errors" (default), "extensions", "disabled"
 ```
 
-By default, when part of a query is filtered by authorization, the list of filtered paths is added to the response and logged by the router. Filtering parts of a query according to the client's rights may be seen as normal operation, and not warrant investigation by platform operators, so those logs can be disabled.
\ No newline at end of file
+By default, when part of a query is filtered by authorization, the list of filtered paths is added to the response and logged by the router. We may need more flexibility in reporting errors, for various reasons:
+- filtering parts of a query according to the client's rights may be seen as normal operation, and not warrant investigation by platform operators, so logs should be disabled
+- when testing authorization, we may not want the clients to raise exceptions if authorization errors are present in the response's `errors` field, so we can put them in `extensions` instead
+- we may not want to give the clients any information about authorization errors, so they can be disabled entirely
```

Comment:
maybe a note about combining `response: extensions` or `response: disabled` with `dry_run: true`?

Reply:
or maybe a non-normative section about the adoption story (add directives, dry_run+extensions/logging, validate behavior, dry_run: false/errors)

Reply:
yup, I think we should have a section on how to migrate easily. But maybe that should live in odyssey?

Reply:
@shorgi ^